commit ac315c398ba671ca33f16b8fa2b73bebd3135b07 (refs/changes/92/2292/3)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-07-26T16:39:19-07:00 (2 years, 2 months ago)
    
    DOCKER-1078 need better logging for docker image create failures
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/endpoints/admin/images-v2.js b/lib/endpoints/admin/images-v2.js
index 6e18bf3..5fb5194 100644
--- a/lib/endpoints/admin/images-v2.js
+++ b/lib/endpoints/admin/images-v2.js
@@ -51,14 +51,18 @@ function adminCreateImageV2(req, res, next) {
         params.owner_uuid = req.app.config.adminUuid;
     }
 
-    ImageV2.create(req.app, req.log, params, function (err, img) {
-        if (err) {
-            next(err);
-        } else {
-            res.send(200, img.serialize());
-            next();
-        }
-    });
+    try {
+        ImageV2.create(req.app, req.log, params, function (err, img) {
+            if (err) {
+                next(err);
+            } else {
+                res.send(200, img.serialize());
+                next();
+            }
+        });
+    } catch (ex) {
+        next(ex);
+    }
 }
 
 
diff --git a/lib/wfapi/workflows/pull-image-v2.js b/lib/wfapi/workflows/pull-image-v2.js
index 1d17dcf..4da3f51 100644
--- a/lib/wfapi/workflows/pull-image-v2.js
+++ b/lib/wfapi/workflows/pull-image-v2.js
@@ -95,6 +95,7 @@ function pullImageLayersV2(job, cb) {
             data.owner_uuid = job.params.account_uuid;
             dockerAdmin.post(path, data, function (adminErr) {
                 if (adminErr) {
+                    queueError = adminErr;
                     job.log.error({config_digest: data.config_digest},
                         'createDockerImage:: error: %s', adminErr);
                 } else {
@@ -153,7 +154,8 @@ function pullImageLayersV2(job, cb) {
                     // If there was no error, yet there was no image created,
                     // then there must have been a timeout (i.e. connection was
                     // closed abruptly), so fire the callback with an error.
-                    cb(new Error('pullImageLayers failed - timeout'));
+                    cb(new Error('pullImageLayers failed - '
+                        + 'no successful create-docker-image'));
                 } else {
                     cb(null, 'pullImageLayers completed');
                 }
